src/app/OperationalDeviceProxy.h

```diff
@@ -72,7 +72,50 @@ struct DeviceProxyInitParams
 
 class OperationalDeviceProxy;
 
-typedef void (*OnDeviceConnected)(void * context, OperationalDeviceProxy * device);
+/**
+ * @brief Helper function as applications refactor away from holding onto a device proxy.
```

Comment:
I think you should first provide the purpose of this object, before providing guidance on when to use it.

Reply:
_"DeviceProxySession is a minimal implementation of DeviceProxy that encapsulates a SessionHolder to track a CASE session. It is meant to provide a transition for existing consumers of OperationalDeviceProxy that were delivered a reference to that object in their respective OnDeviceConnected callback, but were incorrectly holding onto that object pass the function call. DeviceProxySession can be held on for as long as is desired, while still minimizing the code changes needed to transition to a more final solution by virtue of implementing DeviceProxy"_